src/resolve-info.ts

```diff
@@ -0,0 +1,320 @@
+  [fieldName: string]: FieldExpansion | true;
+}
+
+/**
+ * Compute the GraphQLJitResolveInfo's `fieldExpansion` and return a function
+ * that returns the computed resolveInfo. This thunk is registered in
+ * context.dependencies for the field's resolveInfoName
+ */
+export function createResolveInfoThunk({
+  schema,
+  fragments,
+  operation,
+  parentType,
+  fieldName,
+  fieldType,
+  fieldNodes
+}: {
+  schema: GraphQLResolveInfo["schema"];
+  fragments: GraphQLResolveInfo["fragments"];
+  operation: GraphQLResolveInfo["operation"];
+  parentType: GraphQLObjectType;
+  fieldType: GraphQLOutputType;
+  fieldName: string;
+  fieldNodes: FieldNode[];
+}) {
+  const returnType = resolveEndType(fieldType);
+
+  // Result
+  const fieldExpansion: FieldExpansion = {};
+
+  if (returnType != null) {
+    for (const fieldNode of fieldNodes) {
+      handleFieldNode(schema, fragments, returnType, fieldNode, fieldExpansion);
+    }
+  }
+
+  return (
+    rootValue: any,
+    variableValues: any,
+    path: ObjectPath
+  ): GraphQLJitResolveInfo => ({
+    fieldName,
+    fieldNodes,
+    returnType: fieldType,
+    parentType,
+    path,
+    schema,
+    fragments,
+    rootValue,
+    operation,
+    variableValues,
+    fieldExpansion
+  });
+}
+
+type FragmentsType = GraphQLResolveInfo["fragments"];
+type GraphQLNamedOutputType = GraphQLNamedType & GraphQLOutputType;
+
+function handleSelectionSet(
+  schema: GraphQLSchema,
+  fragments: FragmentsType,
+  possibleTypes: GraphQLCompositeType[],
+  selectionSet: SelectionSetNode,
+  fieldExpansion: FieldExpansion
+) {
+  for (const selection of selectionSet.selections) {
+    handleSelection(
+      schema,
+      fragments,
+      possibleTypes,
+      selection,
+      fieldExpansion
+    );
+  }
+}
+
+/**
+ * Compute a list of possible Types from the returnType of the field and
+ * build the TypeExpansion object for the Field. Pass it down to other
+ * handlers to populate.
+ */
+function handleFieldNode(
+  schema: GraphQLSchema,
+  fragments: FragmentsType,
+  returnType: GraphQLOutputType,
+  node: FieldNode,
+  fieldExpansion: FieldExpansion
+) {
+  if (node.selectionSet != null) {
+    const resolvedType = resolveEndType(returnType);
+    const possibleTypes = getPossibleTypes(
+      schema,
+      // if there is a selectionSet, the resolved type must be a composite type
+      resolvedType as GraphQLCompositeType
+    );
+
+    for (const typ of possibleTypes) {
+      if (!Object.prototype.hasOwnProperty.call(fieldExpansion, typ.name)) {
+        fieldExpansion[typ.name] = {};
+      }
+    }
+
+    handleSelectionSet(
+      schema,
+      fragments,
+      possibleTypes,
+      node.selectionSet,
+      fieldExpansion
+    );
+  }
+}
+
+/**
+ * Handle different kinds of selection nodes accordingly.
+ *
+ * For a field, add the field to the TypeExpansion. Create and link
+ * the next FieldExpansion object to be handled recursively
+ *
+ */
+function handleSelection(
```

Comment:
We should file an issue that `@skip` is not supported to be fixed later.

Reply:
https://github.com/ruiaraujo/graphql-jit/issues/17